src/email_service/auto_resume/lambda_function.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import os
+import time
+from typing import Any
+
+import requests
+from sqs import get_sqs_message, send_message_to_queue
+
+# Set up logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Get environment variables
+AUTO_RESUMER_SQS_QUEUE_URL = os.getenv("AUTO_RESUMER_SQS_QUEUE_URL")
+UPSERT_RUN_SQS_QUEUE_URL = os.getenv("UPSERT_RUN_SQS_QUEUE_URL")
+ENVIRONMENT = os.environ.get("ENVIRONMENT")
+DOMAIN_NAME = os.getenv("DOMAIN_NAME")
+
+
+def ensure_database_awake() -> bool:
+    """
+    Call health check API to ensure Aurora Serverless v2 is awake.
+
+    :return: True if database is confirmed awake, False otherwise
+    """
+    health_check_url = f"https://{ENVIRONMENT}-email-service-internal-api-tpet.{DOMAIN_NAME}/{ENVIRONMENT}/email-service/health"
+    max_retries = 10
+    retry_delay = 7  # seconds
+
+    for attempt in range(max_retries):
+        try:
+            logger.info(
+                "Attempting database health check (attempt %d/%d)",
+                attempt + 1,
+                max_retries,
+            )
+            response = requests.get(health_check_url, timeout=5)
+
+            if response.status_code == 200:
+                logger.info("Database confirmed to be awake and healthy")
+                return True
```

Comment:
The auto-resume logic currently checks if the HTTP status code is 200 to determine whether the email service is healthy. However, it should actually check the status field in the response body, and confirm that it equals "HEALTHY"

For details, see: https://awsambassadordevteam.slack.com/archives/C0732CQSDCJ/p1759564017458419?thread_ts=1755528466.318499&cid=C0732CQSDCJ